pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -341,40 +349,34 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 			cidrStrings = append(cidrStrings, ipv6Addr.String())
 		}
 	} else {
-		// multi-networking enabled clusters
-		cidrStrings, err = ca.performMultiNetworkCIDRAllocation(node, instance.NetworkInterfaces)
-		if err != nil {
-			nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
-			return fmt.Errorf("failed to get cidr(s) from provider: %v", err)
+		if hasNodeLabels {
+			cidrStrings = ca.getDefaultNwCIDRs(node, instance.NetworkInterfaces, defaultSubnet, defaultPodRange)
 		}
 	}
-	if len(cidrStrings) == 0 {
-		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
-		return fmt.Errorf("failed to allocate cidr: Node %v has no CIDRs", node.Name)
-	}
-	// Can have at most 2 ips (one for v4 and one for v6)
-	if len(cidrStrings) > 2 {
-		klog.InfoS("Got more than 2 ips, truncating to 2", "cidrStrings", cidrStrings)
-		cidrStrings = cidrStrings[:2]
-	}
```

Comment:
If you keep the performMultiNetworkCIDRAllocation call above then you can keep these checks.

Reply:
I wrapped the cidrStrings check and update in the function below.